src/Orchard.Web/Modules/Orchard.Roles/Conditions/RoleCondition.cs

```diff
@@ -22,15 +22,16 @@ public class RoleConditionProvider : IConditionProvider {
                 return;
             }
 
+            var roles = ((object[])evaluationContext.Arguments).Cast<string>();
             var user = _authenticationService.GetAuthenticatedUser();
             if (user == null) {
                 evaluationContext.Result = false;
-                return;
+                evaluationContext.Result = !string.IsNullOrEmpty(roles.FirstOrDefault(s=>s.Contains("Anonymous")));
```

Comment:
My bad, roles is already a `string[]`, so then just do `evaluationContext.Result = roles.FirstOrDefault(x => String.Equals(x, "Anonymous", StringComparison.OrdinalIgnoreCase))`

Unless `string[].IndexOf` can also do case insensitive checks, then use it.